packages/neuron-ui/src/widgets/DropdownWithCustomRender/index.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { ReactNode, useEffect, useMemo, useRef, useState } from 'react'
```

Comment:
Can we use `packages/neuron-ui/src/widgets/Select/index.tsx` or `packages/neuron-ui/src/widgets/Dropdown/index.tsx`?

Reply:
> Can we use `packages/neuron-ui/src/widgets/Select/index.tsx` or `packages/neuron-ui/src/widgets/Dropdown/index.tsx`?

Just to be different from the original dropdown, because this is a new component and the original dropdown may be used elsewhere.

I can change the name of this componnet.

Reply:
It seems both the `DropDown` and `Select` are in use in some other components, so I'll still call it by this name.